lib/shopping_cart.rb

```diff
@@ -0,0 +1,51 @@
+class ShoppingCart
+    attr_reader :name, :capacity , :products, :details
+    def initialize(name, capacity, products = [])
+        @name = name
+        @capacity = capacity.to_i
+        @products = products
+        @details = {name: @name,  capacity: @capacity.to_i, }
+    end
+
+    def add_product(product)
+        products << product
+    end
+
+    def calculate_total
+        total = 0
+        @products.each do |product|
+            total += product.quantity
+        end
+        total
+    end
+
+    def total_number_of_products
+        calculate_total
```

Comment:
interesting that you're calling the method, why not put the code from lines 15 through 19 in this method? I think that's what I'd suggest doing.. especially since you're calling total_number_of_products in is_full?